packages/test/test-end-to-end-tests/src/test/gc/gcSweepAttachmentBlobs.spec.ts

```diff
@@ -1086,4 +1126,246 @@ describeCompat("GC attachment blob sweep tests", "NoCompat", (getTestObjectProvi
 			validateBlobStateInSummary(summary3.summaryTree, blob1NodePath);
 		});
 	});
+
+	describe("Sweep with summarize failures and retries", () => {
+		const summarizeErrorMessage = "SimulatedTestFailure";
+		/**
+		 * This function does the following:
+		 * 1. Overrides the summarize function of the given container runtime to fail until final summarize attempt.
+		 *
+		 * 2. If "pauseInbound" is true, pauses the inbound queue until the final summarize attempt is completed.
+		 *
+		 * 3. Generates and returns a promise which resolves with ISummarizeEventProps on successful summarization.
+		 */
+		async function overrideSummarizeAndGetCompletionPromise(
+			summarizer: ISummarizer,
+			containerRuntime: ContainerRuntime,
+			pauseInbound: boolean = false,
```

Comment:
Default of true would match typical production behavior, right? Or this is saying keep it paused even in between attempts?  Then never mind.

Reply:
Maybe rename it to `blockInboundGcOp` or something?

Reply:
renamed